### ISSUE TITLE
Elements in a task list should be moved when they are checked.

### ISSUE THREAD

**Issue by Domant3lis** [nextcloud/notes#681]
# The problem:
It has become really annoying when checked elements mix in with unchecked elements in a task list.

# The solution:
Let's say we have a list:
- [ ] Task 4
- [ ] Task 3
- [x] Task 2
- [x] Task 1

Now if, for example, Task 4 is checked it should be automatically moved above Task 2 but bellow Task 3, like this:
- [ ] Task 3
- [x] Task 4
- [x] Task 2
- [x] Task 1

The opposite should also be true. When An element is unchecked it should be moved up. For example if Task 1 is unchecked it should appear bellow Task 3 but above Task 4, like this:
- [ ] Task 3
- [ ] Task 1
- [x] Task 4
- [x] Task 2

Basically, that I proposed is automatic sorting of checked and unchecked elements in a list.
This way checked elements stay out of the way and do not clutter the list.

**Comment by stefan-niedermann** [comment]
I disagree for the reasons i also [wrote for the Android Notes app](https://github.com/stefan-niedermann/nextcloud-notes/pull/987#issuecomment-735442768).

**Comment by korelstar** [comment]
I agree with @stefan-niedermann. We should not try to interpret the order of list items. Some people will like an auto-sorting feature, but there are definitely people who will hate such a feature, because they want to keep the manual order.
